### ISSUE TITLE
Error when trying to get config

### ISSUE THREAD

**Issue by webpro** [release-it/release-it#861]
I got an error when try to get config.

```
Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './lib/config' is not defined by "exports" in /Users/jairs/dev/boti/news-fragments/node_modules/release-it/package.json imported from /Users/jairs/dev/boti/news-fragments/src/config.js
    at new NodeError (node:internal/errors:371:5)
    at throwExportsNotFound (node:internal/modules/esm/resolve:429:9)
    at packageExportsResolve (node:internal/modules/esm/resolve:683:3)
    at packageResolve (node:internal/modules/esm/resolve:853:14)
    at moduleResolve (node:internal/modules/esm/resolve:910:18)
    at defaultResolve (node:internal/modules/esm/resolve:1005:11)
    at ESMLoader.resolve (node:internal/modules/esm/loader:475:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:245:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:79:40)
    at link (node:internal/modules/esm/module_job:78:36) {
  code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
}
```

https://github.com/grupoboticario/news-fragments/blob/main/src/config.js#L1

_Originally posted by @jairhenrique in https://github.com/release-it/release-it/issues/859#issuecomment-1016532554_

**Comment by webpro** [comment]
The idea is to get this config from the plugin, like so:

```js
class MyPlugin extends Plugin {
  init() {
    const config = this.config.getContext();
  }
};
```

**Comment by jairhenrique** [comment]
@webpro when I try to run my tests, `this.config` is `undefined`

**Comment by ath0mas** [comment]
starting with esm, I have similar issue with `import { hasAccess } from 'release-it/lib/util';`
(previously `const { hasAccess } = require('release-it/lib/util');`)

results in **Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './lib/util' is not defined by "exports" …**

**Comment by jairhenrique** [comment]
@webpro what is your suggestion to solve the situation of my plugin? Today I use it in the release process itself.

```js
"release-it": {
    "npm": false,
    "hooks": {
      "before:init": "npm test",
      "after:bump": "node --experimental-json-modules myself-release.js"
    },
    "plugins": {
      "@release-it/bumper": {
        "out": [
          {
            "file": "sonar-project.properties",
            "type": "text/x-properties",
            "path": "sonar.projectVersion"
          },
          {
            "file": "package.json",
            "path": "version"
          }
        ]
      }
    }
  },
```

```
"after:bump": "node --experimental-json-modules myself-release.js"
```

[myself-release.js](https://github.com/grupoboticario/news-fragments/blob/main/myself-release.js)

**Comment by webpro** [comment]
@jairhenrique When I add something like this in a plugin, things seem fine during test runs:

```
class ConventionalChangelog extends Plugin {
  init() {
    console.log(this.config.getContext());
  }
```

Maybe you could share the situation in which this isn't working properly?

**Comment by webpro** [comment]
> starting with esm, I have similar issue with `import { hasAccess } from 'release-it/lib/util.js';` (previously `const { hasAccess } = require('release-it/lib/util');`)
> 
> based on same `import` migrated for _npm_ plugin : see [npm.js#L5](https://github.com/release-it/release-it/blob/feature/es-modules/lib/plugin/npm/npm.js#L5)
> 
> results in **Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './lib/util' is not defined by "exports" …**
> 
> Or how to use same libs and utils as any other internal plugins?

The internal plugins aren't meant to be imported in the first place, as they're not part of the public API and could change any time without notice (and without a major version bump).

As an alternative you could simply copy-paste the code you need.

**Comment by webpro** [comment]
You can keep/set release-it v14 as a `peerDependency` for any plugin. And migrate if/when you want :)

Also happy to see how things could be improved on release-it and/or plugin side if possible.

**Comment by markcellus** [comment]
Thanks for opening this issue @webpro . For our case, we have a tool that wraps release-it and we need to get the release-it config from the consumer's root (local). This is what we're doing and getting this same error with v15:

```js
const Config = require('release-it/lib/config');
const { localConfig } = new Config();
// do stuff with `localConfig`
```

I'm not sure if this is the best practice but its the only way we can get the consumer's local release-it configuration. Also open to any advice or recommendations. Thanks again.

**Comment by markcellus** [comment]
Is there a workaround for this one?

**Comment by webpro** [comment]
The `Config` class and especially the `localConfig` property are implementation details that might change.

Isn't it possible to load the JSON (or whatever) config and pass it to the main release-it function?

```js
import releaseIt from 'release-it';
const config = JSON.parse(await fs.readFile('release-it.json'));
releaseIt(config);
```

**Comment by markcellus** [comment]
@webpro The reason we're using `Config` from this package is because we need to read from all possible config file formats (ie js, json, etc). Using localConfig is the safest way. Your example wouldn't work if consumers are using .release-it.json for instance.

**Comment by webpro** [comment]
Extracting just this part out of the `Config` module, this is currently in release-it to populate `localConfig`:

```
import { cosmiconfigSync } from 'cosmiconfig';
import parseToml from '@iarna/toml/parse-string.js';

const searchPlaces = [
  'package.json',
  '.release-it.json',
  '.release-it.js',
  '.release-it.cjs',
  '.release-it.yaml',
  '.release-it.yml',
  '.release-it.toml'
];

const loaders = {
  '.toml': (_, content) => parseToml(content)
};

const getLocalConfig = () => {
  const explorer = cosmiconfigSync('release-it', {
    searchPlaces,
    loaders
  });
  const result = explorer.search();
  return result.config;
};
```

Alternatively, we can export the `Config` class (like `Plugin`). I just released [15.1.0-next.0](https://github.com/release-it/release-it/releases/tag/15.1.0-next.0)

```
npm install release-it@next
```

Does that help/work?

**Comment by markcellus** [comment]
Would rather avoid duplicating code in case new config file formats are changed in the future. 

I tried with 15.1.0-next.0 but it appears that v15+ versions all assume ESM. So getting errors like this now:

```
Error [ERR_REQUIRE_ESM]: Must use import to load ES Module: /code/rally/release/node_modules/release-it/lib/index.js
require() of ES modules is not supported.
require() of /code/rally/release/node_modules/release-it/lib/index.js from /code/rally/release/src/file.js is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules.
Instead rename index.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from /code/rally/release/node_modules/release-it/package.json.

```

**Comment by webpro** [comment]
This is released in [v15.1.0](https://github.com/release-it/release-it/releases/tag/15.1.0)
